# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 40031ad88dbd
# Full Hash: 40031ad88dbd9af3296493ae4bf501c158cb0ca2
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r cee7eaa8c402 -r 40031ad88dbd js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Tue Nov 03 10:25:14 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Tue Nov 03 10:28:10 2020 +0000
@@ -4415,12 +4415,11 @@
   explicit WasmABIArgIter(const VecT& types)
       : ABIArgIterBase<VecT, WasmABIArgGenerator>(types) {}
 };
-
 }  // namespace jit
 
 namespace wasm {
-TlsData* ExtractCalleeTlsFromFrameWithTls(Frame* fp);
-TlsData* ExtractCallerTlsFromFrameWithTls(Frame* fp);
+const TlsData* ExtractCalleeTlsFromFrameWithTls(const Frame* fp);
+const TlsData* ExtractCallerTlsFromFrameWithTls(const Frame* fp);
 }  // namespace wasm
 
 }  // namespace js